### PR TITLE
Fixes `check_coords` function calls inside `for` loops and `check_pres_units` return

### LINKED ISSUE #38: Small bug in check_pres_units
Thanks, this package is really useful! I encountered a small bug in the function check_pres_units -- if units aren't given in the array, then da isn't returned and no array is output which can cause problems.

### LINKED ISSUE #41: Unassigned result of `check_coords` function in feedback calculations
Hi, 
I noticed that the `check_coords` function calls located inside `for` loops do not actually adapt the coordinates as intended by the function. In the loops referenced below, you are only updating the local variable `d`, but not to the actual feedback variable. To fix this, you need to explicitly assign the result back to the variables (like in e.g. https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L211C9-L211C44)

Issues:

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L75C1-L76C28

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L193C1-L197C28

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L366C1-L370C1

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L866C1-L867C36

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L1006C1-L1009C28

https://github.com/tyfolino/climkern/blob/b471b5dd724e7e31077b26fa424090ebd4fb37b6/climkern/frontend.py#L1193C1-L1196C28

### PR DESCRIPTION
**Pull request by olinke95**
Solves issues #41 and #38, respectively

### CODE REVIEW

File: climkern/frontend.py [hunk 1]
```diff
@@ -72,9 +72,11 @@ def calc_alb_feedback(
     alb_key = "sw_a" if check_sky(sky) == "all-sky" else "swclr_a"
 
     # check input coordinates
-    for d in [ctrl_rsus, ctrl_rsds, pert_rsus, pert_rsds]:
```
Comment (tyfolino) [inline climkern/frontend.py:75]: Hah, thanks for catching this error. Even after using Python for 10 years, I still make mistakes like this!

**Review by tyfolino — APPROVED** [review]
I tested this using built-in tests and produced a few edge cases to make sure this is working properly. Thanks for contributing this!
